tests/baselines/reference/assignmentCompatWithGenericCallSignaturesWithOptionalParameters.types

```diff
@@ -481,11 +481,11 @@ module GenericSignaturesInvalid {
 >b.a2 = t.a : () => T
 >           : ^^^^^^^
 >b.a2 : <T_1>(x?: T_1) => T_1
->     : ^   ^^ ^^^   ^^^^^^^^
+>     : ^^^^^^ ^^^^^^^^^^^^^^

```

Comment:
Same here; tihs looks worse? Though, `T_1` wasn't in the original source, so I guess this is just a baseline fix?

Reply:
I found the issue. I was looking up the original parameter name `T` to make sure it is available in the scope the signature is used in. The problem is the node was already renamed to prevent shadowing (so I was looking for `T` instead of `T_1`). This caused the whole signature to be thrown asway and fall back to regular type printing.

Reply:
I think I'm half confused because if it's generated, doesn't that mean none of that should be reused? I thought the point of the underlining is to show what does and doesn't come from the original file?

Reply:
I guess it depends what you call reuse. Renaming type parameters to avoid shadowing is still considered reuse. It does use the same structure as the original source just with a different parameter name. That logic was already there maybe @weswigham  has some more thoughts.

Reply:
After Wesley's explanation in the ID meeting this morning, this seems fine if the definition of reuse is not as strict as I expected.